---
title: "Creating a Query Gene Signature for Triple-Negative Breast Cancer from TCGA data"
author: Mr Michael J. Jones
output: html_document
---

## Introduction

In order to perform Gene Expression Connectivity Map, we first require a Query Gene Signature which 
provides a ranked list of genes or probe IDs that represent significantly up-regulated or down-regulated 
genes between a diseased and a control state. In this case, we wish to create a query gene signature to 
describe significantly differentially expressed genes in Triple-Negative Breast Cancer.

Because of the heterogeneity of Triple-Negative Breast Cancer, subsequently, we would like to create a set 
of query gene signatures, each for one of six different subtypes of the disease identified previously by Lehmann et al.

<!-- Add reference somehow -->
## Packages Used

Firstly, let us create this convenience function to conditionally install packages we need.

```{r}
runIfNotInstalled <- function(packageName, fun) {
  if (! packageName %in% installed.packages()) {
    fun()
  }
}
```

Data is collected from 'The Cancer Genome Atlas' using the bioconductor package `TCGAbiolinks`. We will use this package 
to download RNA-seq data for Triple-Negative Breast Cancer and normal tissue. This package has the added benefit of being able of being able to perform differntial expression analysis by itself.

```{r importtcga}
source("http://bioconductor.org/biocLite.R")


runIfNotInstalled("TCGAbiolinks", function() biocLite("TCGAbiolinks"))
library("TCGAbiolinks")

runIfNotInstalled("SummarizedExperiment", function() biocLite("SummarizedExperiment"))
library("SummarizedExperiment")

```



## Analysis

The analysis below is derived 'TCGA Downstream Analysis: Case Studies' from the bioconductor documentation for TCGABiolinks. <https://bioconductor.org/packages/devel/bioc/vignettes/TCGAbiolinks/inst/doc/tcgaBiolinks.html>

```{r downloadbrca}
query <- GDCquery(project = "TCGA-BRCA",
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  workflow.type = "HTSeq - FPKM-UQ"
) 
GDCdownload(query, chunks.per.download = 100)

data <- GDCprepare(query)
```

This may take some time to download.

Now we can retrieve subtype information

```{r subtypeinfo}
dataSubt <- TCGAquery_subtype(tumor = "BRCA")
```

as well as the clinical data
```{r clinicaldata}
dataClin <- GDCquery_clinic(project = "TCGA-BRCA","clinical") 
```

## Filtering out the Triple-Negative Breast Cancer data
Now that we have all Breast Cancer TCGA data, we need to filter it such that we are left with the Triple-Negative Breast Cancer data.

```{r filtertnbc}

er.negative.status <- !is.na(data@colData$subtype_ER.Status) & data@colData$subtype_ER.Status == 'Negative'
pr.negative.status <- !is.na(data@colData$subtype_PR.Status) & data@colData$subtype_PR.Status == 'Negative'
her2.negative.status <- !is.na(data@colData$subtype_HER2.Final.Status) &
  data@colData$subtype_HER2.Final.Status == 'Negative'

tnbc.status <- er.negative.status & pr.negative.status & her2.negative.status

```

This leaves us with `r length(which(tnbc.status))` TNBC profiles. Let us filter out this data so we are only
analysing the TNBC profiles.

```{r tnbcdata}
tnbc.data <- data[which(tnbc.status),]
```

## Analysis
<!-- TODO make sure only TNBC data being analysed -->

Here, we separate the data into both Solid Tumour (TP) and Normal Tissue (NT) for differential expression analysis
```{r}
dataSmTP <- TCGAquery_SampleTypes(colnames(tnbc.data), "TP")
dataSmNT <- TCGAquery_SampleTypes(colnames(tnbc.data), "NT")
```

Now we may perform the differential expression analysis
```{r}
# Prepare expression matrix with geneID in the rows and samples (barcode) in the columns
TNBCMatrix <- assay(tnbc.data)

dataDEGs <- TCGAanalyze_DEA(mat1 = TNBCMatrix[,dataSmNT],
                            mat2 = TNBCMatrix[,dataSmTP],
                            Cond1type = "Normal",
                            Cond2type = "Tumor",
                            fdr.cut = 0.01 ,
                            logFC.cut = 1,
                            method = "glmLRT")  
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
