---
title: Creating a Query Gene Signature for Triple-Negative Breast Cancer from TCGA
  data
author: "Mr Michael J. Jones"
output: pdf_document
---

## Introduction

In order to perform Gene Expression Connectivity Map, we first require a Query Gene Signature which 
provides a ranked list of genes or probe IDs that represent significantly up-regulated or down-regulated 
genes between a diseased and a control state. In this case, we wish to create a query gene signature to 
describe significantly differentially expressed genes in Triple-Negative Breast Cancer.

Because of the heterogeneity of Triple-Negative Breast Cancer, subsequently, we would like to create a set 
of query gene signatures, each for one of six different subtypes of the disease identified previously by Lehmann et al.

<!-- Add reference somehow -->
## Packages Used

Firstly, let us create this convenience function to conditionally install packages we need.

```{r}
runIfNotInstalled <- function(packageName, fun) {
  if (! packageName %in% installed.packages()) {
    fun()
  }
}
```

Data is collected from 'The Cancer Genome Atlas' using the bioconductor package `TCGAbiolinks`. We will use this package 
to download RNA-seq data for Triple-Negative Breast Cancer and normal tissue. This package has the added benefit of being able of being able to perform differntial expression analysis by itself.

```{r importtcga}
source("http://bioconductor.org/biocLite.R")


runIfNotInstalled("TCGAbiolinks", function() biocLite("TCGAbiolinks"))
library("TCGAbiolinks")

runIfNotInstalled("SummarizedExperiment", function() biocLite("SummarizedExperiment"))
library("SummarizedExperiment")

```



## Analysis

The analysis below is derived 'TCGA Downstream Analysis: Case Studies' from the bioconductor documentation for TCGABiolinks. <https://bioconductor.org/packages/devel/bioc/vignettes/TCGAbiolinks/inst/doc/tcgaBiolinks.html>.

The data we download is `FPKM-UQ`, meaning that it's upper-quartile normalized so that 

```{r downloadbrca}
query <- GDCquery(project = "TCGA-BRCA",
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  workflow.type = "HTSeq - FPKM-UQ"
)

data.already.downloaded <- file.exists("GDCdata/")
if (!data.already.downloaded) GDCdownload(query, chunks.per.download = 100)

data <- GDCprepare(query)
```

This may take some time to download.

Now we can retrieve subtype information

```{r subtypeinfo}
dataSubt <- TCGAquery_subtype(tumor = "BRCA")
```

as well as the clinical data
```{r clinicaldata}
dataClin <- GDCquery_clinic(project = "TCGA-BRCA","clinical") 
```

## Filtering out the Triple-Negative Breast Cancer data
Now that we have all Breast Cancer TCGA data, we need to filter it such that we are left with the Triple-Negative Breast Cancer data.

```{r filtertnbc}

er.negative.status <- !is.na(data@colData$subtype_ER.Status) & data@colData$subtype_ER.Status == 'Negative'
pr.negative.status <- !is.na(data@colData$subtype_PR.Status) & data@colData$subtype_PR.Status == 'Negative'
her2.negative.status <- !is.na(data@colData$subtype_HER2.Final.Status) &
  data@colData$subtype_HER2.Final.Status == 'Negative'

tnbc.status <- er.negative.status & pr.negative.status & her2.negative.status

```

This leaves us with `r length(which(tnbc.status))` TNBC profiles. Let us filter out this data so we are only
analysing the TNBC profiles.

```{r tnbcdata}
tnbc.data <- data[which(tnbc.status),]
```

## Analysis

Here, we separate the data into both Solid Tumour (TP) and Normal Tissue (NT) for differential expression analysis
```{r}
dataSmTP <- TCGAquery_SampleTypes(colnames(tnbc.data), "TP")
dataSmNT <- TCGAquery_SampleTypes(colnames(tnbc.data), "NT")
```

<!-- 
  TODO look at vignette for TCGAbiolinks - do I need to normalize data? 
  Ask Shu-Dong, perhaps
-->

Now we may perform the differential expression analysis.
```{r}
# Prepare expression matrix with geneID in the rows and samples (barcode) in the columns
TNBCMatrix <- assay(tnbc.data)

dataDEGs <- TCGAanalyze_DEA(mat1 = TNBCMatrix[,dataSmNT],
                            mat2 = TNBCMatrix[,dataSmTP],
                            Cond1type = "Normal",
                            Cond2type = "Tumor",
                            fdr.cut = 0.01 ,
                            logFC.cut = 1,
                            method = "glmLRT") 

knitr::kable(dataDEGs, format = "markdown")
```

## Create TNBC Query Gene Signature

In order for us to create a Query Gene Signature that's compatible with the the reference profiles, we need 
to convert the ensembl IDs into the equivalent probe IDs from the Affymetrix HG-U133A microarray platform.

To do this, we use the `biomaRt` package
```{r installbiomart}
runIfNotInstalled("biomaRt", function() biocLite("biomaRt"))
library("biomaRt")
ids <- rownames(dataDEGs)
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
summary <- getBM(filters = 'ensembl_gene_id', attributes= 
                   c("ensembl_gene_id", "affy_hg_u133_plus_2", "description"),
                 values=ids,mart= mart)

```

There may be cases where there are multiple probe IDs for each ensign gene id. Let's remove duplicates:

```{r removeduplicateids}
non.duplicate.rows <- which(!duplicated(summary$ensembl_gene_id))
non.duplicate.ensembl.ids <- summary[non.duplicate.rows,]
```

and we can append these data to our differential expression analysis `data.frame`
```{r appendbiomarttodegs}
data.degs.with.affy.probes <- data.frame(non.duplicate.ensembl.ids, dataDEGs)
knitr::kable(data.degs.with.affy.probes, format = "markdown")
```

Next, we must remove the records without corresponding probe IDs for each ensembl gene.

```{r removeblankprobe}
non.blank.probe.ids <- !is.na(data.degs.with.affy.probes$affy_hg_u133_plus_2) & data.degs.with.affy.probes$affy_hg_u133_plus_2 != ""
data.degs.non.empty.affy <- data.degs.with.affy.probes[which(non.blank.probe.ids),]
```

Next we will create the unordered query gene signature by taking the signs of each of the `logFC` and making them `-1` or `+1`

```{r unorderedqgs}
data.degs.non.empty.affy <- data.frame(data.degs.non.empty.affy, unordered=sign(data.degs.non.empty.affy$logFC))
```

And to create the orderd query gene signature we multiply the rank of each logFC by the unordered query gene signature.

```{r orderedqgs}
fold.change <- data.degs.non.empty.affy$logFC

ranking <- list(index=order(decreasing=T,abs(fold.change)), rank=1:length(fold.change))

order.by.fc <- data.degs.non.empty.affy[ranking$index,]
data.degs.non.empty.affy <- data.frame(order.by.fc, ordered=order.by.fc$unordered * 1:length(fold.change))

resulting.qgs <- data.frame(probe.id=data.degs.non.empty.affy$affy_hg_u133_plus_2, unordered=data.degs.non.empty.affy$unordered, ordered=data.degs.non.empty.affy$ordered)
knitr::kable(resulting.qgs, format = "markdown")
```


<!-- 
  TODO find Gene Ontology for pathway maybe
  TODO split up TNBC into subtypes and create Query Gene Signatures for those
-->

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
